rllib/evaluation/collectors/agent_collector.py

```diff
@@ -553,9 +565,13 @@ def _build_buffers(self, single_row: Dict[str, TensorType]) -> None:
             should_flatten_action_key = (
                 col == SampleBatch.ACTIONS and not self.disable_action_flattening
             )
+
+            should_flatten_state_key = (
+                col.startswith("state_out") and not self._enable_rl_module_api
+            )
```

Comment:
This is the same as before, only that it does not include RL Modules because states there need no flattening.